tests/components/seventeentrack/test_sensor.py

```diff
@@ -0,0 +1,327 @@
+from homeassistant.util import utcnow
+from tests.common import get_test_home_assistant, MockDependency, \
+    get_test_config_dir, fire_time_changed
+
+VALID_CONFIG_MINIMAL = {
+    'sensor': {
+        'platform': 'seventeentrack',
+        CONF_USERNAME: 'test',
+        CONF_PASSWORD: 'test'
+    }
+}
+
+INVALID_CONFIG = {
+    'sensor': {
+        'platform': 'seventeentrack',
+        'boom': 'test',
+    }
+}
+
+VALID_CONFIG_FULL = {
+    'sensor': {
+        'platform': 'seventeentrack',
+        CONF_USERNAME: 'test',
+        CONF_PASSWORD: 'test',
+        CONF_SHOW_ARCHIVED: True,
+        CONF_SHOW_DELIVERED: True
+    }
+}
+
+VALID_CONFIG_FULL_NO_DELIVERED = {
+    'sensor': {
+        'platform': 'seventeentrack',
+        CONF_USERNAME: 'test',
+        CONF_PASSWORD: 'test',
+        CONF_SHOW_ARCHIVED: False,
+        CONF_SHOW_DELIVERED: False
+    }
+}
+
+DEFAULT_SUMMARY = {
+    "Not Found": 0,
+    "In Transit": 0,
+    "Expired": 0,
+    "Ready to be Picked Up": 0,
+    "Undelivered": 0,
+    "Delivered": 0,
+    "Returned": 0
+}
+
+NEW_SUMMARY_DATA = {
+    "Not Found": 1,
+    "In Transit": 1,
+    "Expired": 1,
+    "Ready to be Picked Up": 1,
+    "Undelivered": 1,
+    "Delivered": 1,
+    "Returned": 1
+}
+
+
+class ClientMock:
+    """Mock the py17track client to inject the ProfileMock."""
+
+    def __init__(self, websession) -> None:
+        """Mock the profile."""
+        self.profile = ProfileMock()
+
+
+class ProfileMock:
+    """ProfileMock will mock data coming from 17track."""
+
+    package_list = []
+    login_result = True
+    summary_data = DEFAULT_SUMMARY
+    account_id = '123'
+
+    @classmethod
+    def reset(cls):
+        """Reset data to defaults."""
+        cls.package_list = []
+        cls.login_result = True
+        cls.summary_data = DEFAULT_SUMMARY
+        cls.account_id = '123'
+
+    def __init__(self) -> None:
+        """Override Account id."""
+        self.account_id = self.__class__.account_id
+
+    async def login(self, email: str, password: str) -> bool:
+        """Login mock."""
+        return self.__class__.login_result
+
+    async def packages(self, package_state: Union[int, str] = '',
+                       show_archived: bool = False) -> list:
+        """Packages mock."""
+        return self.__class__.package_list[:]
+
+    async def summary(self, show_archived: bool = False) -> dict:
+        """Summary mock."""
+        return self.__class__.summary_data
+
+
+@pytest.fixture(autouse=True, name="mock_py17track")
+def fixture_mock_py17track():
+    """Mock py17track dependency."""
+    with MockDependency('py17track'):
+        yield
+
+
+@pytest.fixture(autouse=True, name="mock_client")
+def fixture_mock_client(mock_py17track):
+    """Mock py17track client."""
+    with mock.patch('py17track.Client', new=ClientMock):
+        yield
+    ProfileMock.reset()
+
+
+@pytest.fixture(name="hass")
+def fixture_hass(mock_client):
```

Comment:
We don't need a new fixture for hass. It's already available.

Reply:
Also tried with the "stock" hass, but all tests go into timeout:
```
8.99s call     tests/components/seventeentrack/test_sensor.py::test_full_valid_config[pyloop]
0.05s teardown tests/components/seventeentrack/test_sensor.py::test_full_valid_config[pyloop]
0.01s setup    tests/components/seventeentrack/test_sensor.py::test_full_valid_config[pyloop]
```

while logs look good (I can see my entities are created, and so one).
Error says it blocked in `assert setup_component(hass, 'sensor', VALID_CONFIG_FULL)` and actually it's in `/usr/lib/python3.6/threading.py:295` which is `waiter.acquire()` and I get `Failed: Timeout >9.0s`

I don't have the problem with the new fixture. I'm gonna have a deeper look

Reply:
Try using coroutines for the test functions instead.

Reply:
moving all the way to `async` does the trick, thanks !

I just pushed, I think everything should be ok